### PR TITLE
feat: system prompt

### LINKED ISSUE #443: Add support to pass system prompt
### 🚀 The feature

OpenAI supports passing system prompt. We need to add support for this in embedchain

### Motivation, pitch

System prompts help in better controlling the persona of the bot.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cachho**
## Description

Allow passing of a system prompt.

Fixes #443 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [X] Unit Test

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

File: docs/advanced/query_configuration.mdx [hunk 1]
```diff
@@ -65,6 +65,8 @@ _coming soon_
 |top_p|Controls the diversity of words. Higher values (closer to 1) make word selection more diverse, lower values make words less diverse.|float|1|
 |history|include conversation history from your client or database.|any (recommendation: list[str])|None|
 |stream|control if response is streamed back to the user.|bool|False|
+|deployment_name|t.b.a.|str|None|
+|system_prompt|System prompt string. Unused if none.|str|None|
```
Comment (taranjeet) [inline docs/advanced/query_configuration.mdx:69]: In the docs, can you add an example of how to use this actually?

File: embedchain/apps/CustomApp.py [hunk 2]
```diff
@@ -133,15 +133,18 @@ def _get_azure_openai_answer(prompt: str, config: ChatConfig) -> str:
...
         return chat(messages).content
 
     @staticmethod
-    def _get_messages(prompt: str) -> List[BaseMessage]:
+    def _get_messages(prompt: str, system_prompt: Optional[str] = None) -> List[BaseMessage]:
         from langchain.schema import HumanMessage, SystemMessage
 
-        return [SystemMessage(content="You are a helpful assistant."), HumanMessage(content=prompt)]
+        return [
+            SystemMessage(content=system_prompt if system_prompt else "You are a helpful assistant."),
```
Comment (taranjeet) [inline embedchain/apps/CustomApp.py:145]: We should only add if system prompt is present.
Can you update?

File: docs/advanced/query_configuration.mdx (hunk 1 above)
Comment (cachho) [reply docs/advanced/query_configuration.mdx:69]: added example: https://github.com/embedchain/embedchain/pull/448/files#diff-ceb7311e70d50f7fa3a14f5e727a03781b2a4c20c3e8105f3459d1d8af8f1160R71

File: embedchain/apps/CustomApp.py (hunk 2 above)
Comment (cachho) [reply embedchain/apps/CustomApp.py]: fixed

**Comment by taranjeet** [comment]
Looks good to me, merging this now.

One feedback is to have support for system prompt at App init level. Lets say somebody is creating a Harry Potter like bot, in that case, the user would prefer setting it once at the app level only and not each time during query/chat level.

Will open a separate issue for this.

**Review by taranjeet — APPROVED** [review]

**Pull request merged**